src/MongoDB/BulkWriteCommand.stub.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @generate-class-entries static
+ * @generate-function-entries static
+ */
+
+namespace MongoDB\Driver;
+
+/** @not-serializable */
+final class BulkWriteCommand implements \Countable
+{
+    final public function __construct(?array $options = null) {}
+
+    // TODO: decide if Countable is still desirable
+    public function count(): int {}
+
+    final public function deleteOne(string $namespace, array|object $filter, ?array $options = null): void {}
+
+    final public function deleteMany(string $namespace, array|object $filter, ?array $options = null): void {}
+
+    final public function insertOne(string $namespace, array|object $document): mixed {}
+
+    final public function replaceOne(string $namespace, array|object $filter, array|object $replacement, ?array $options = null): void {}
+
+    final public function updateOne(string $namespace, array|object $filter, array|object $update, ?array $options = null): void {}
+
+    final public function updateMany(string $namespace, array|object $filter, array|object $update, ?array $options = null): void {}
+
+    final public function execute(Manager|Server $managerOrServer): mixed {}
```

Comment:
By placing the execute method on this side, it becomes essential to be able to mock this class. But this class is `final`.

How do you test a function that creates a BulkWriteCommand and executes it?

```php
function saveDocuments(Manager $manager, array $documents) {
    $command = new BulkWriteCommand();
    foreach ($documents as $document) {
        $command->replaceOne(/** do something very domain specific */);
    }
    // I have no idea how to track the call to $manager
    $command->execute($manager);

    // This implementation would be easier to mock
    $manager->execute($command);
}
```

Reply:
Thanks for the feedback here. I clearly don't deal with mocking objects anymore, so I overlooked how this would hinder that. I suppose that takes us back to `Manager::executeBulkWriteCommand()`, and a corresponding Server method.

Reply:
Revisiting this since we discussed it earlier in the week with @alcaeus. In terms of mocking, having `execute()` here isn't much different than the Manager and Server classes, which are also both `final`.

We also discussed having `execute()` methods oh other objects, and I believe only Command would be problematic since that currently relies on several variations (read, read/write, write) and to replace that with a singular `execute()` method would require PHPC to inspect command documents.

Anyway, I'll going to proceed with `BulkWriteCommand::execute()` for now and we can revisit later if needed.